### DISCUSSION TITLE
lock.json work failed when the project has one file on CDN

### DISCUSSION THREAD

**Discussion by jiawei397** [denoland/deno#12955] [Q&A] [answered]
Recently, I promoted deno within the team, but I encountered such a problem when creating a template project. 

I hope `lock.json` can be used in my project to ensure the version of the referenced code, but there is a file from CDN. Even if we lock the version number, its hash value will change from time to time. 

First time it maybe:
```json
{
  "https://cdn.skypack.dev/-/bson@v4.5.3-mUXcu8gnafOqHTcdyS0d/dist=es2019,mode=imports/optimized/bson.js": "a3382aaffe1a6a7930a06602192a47c5c83fa3de3090b5d7ff9b6af85b36a50d",
  "https://cdn.skypack.dev/bson@v4.5.3": "a1d77618b8912305def6956cf85f166e66e33de7ab65caeee8001a173d807809"
}
```
But it second time maybe:
```json
{
  "https://cdn.skypack.dev/-/bson@v4.5.3-qyhgJPJ3fRFU6pNRUV8E/dist=es2019,mode=imports/optimized/bson.js": "a3382aaffe1a6a7930a06602192a47c5c83fa3de3090b5d7ff9b6af85b36a50d",
  "https://cdn.skypack.dev/bson@v4.5.3": "4c2fd52b1a47f88905aa92bdf4c7d82a2c55c1f6f652d29a17c849c3debed0ab"
}
```
I have to give it up now. How can I solve this problem?

**Comment by kitsonk** [comment, marked as answer]
You should consider using pinned URLs with with skypack if you are using a lock file.  See there documentation here: https://docs.skypack.dev/skypack-cdn/api-reference/pinned-urls-optimized.

**Reply by jiawei397** [reply]
Thank you for replying to me so soon. This seems to be a good solution. 
I am ready to put forward this improvement proposal to the author of the third-party library. 
Thanks again
